### PR TITLE
[WIP] Improve pinning

### PR DESCRIPTION
**Pull request by JonasBak**
The current pinning system is pretty bad.

Here are some ideas to make it better:
 * Pin events and articles for groups
 * Pin events and articles between two dates
 * Give a better overview of which items are pinned and when

Things that need to be done:
 * Tie it together with frontpage
 * Permissions :100: 
 * Tests :1st_place_medal: 
 * Frontend :rice_ball:

### CODE REVIEW

File: lego/apps/pinned/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
[the remark this answers is not in the document]
Comment (SmithPeder) [reply lego/apps/pinned/admin.py]: :)

[the remark this answers is not in the document]
Comment (JonasBak) [reply lego/apps/pinned/admin.py]: ez django-admin startapp :kissing_heart:

File: lego/apps/pinned/models.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from lego.apps.users.models import AbakusGroup
+from lego.utils.models import BasisModel
+
+
+class Pinned(BasisModel):
+    pinned_from = models.DateField()
+    pinned_to = models.DateField()
+
+    event = models.ForeignKey(
+        Event, related_name='pins', on_delete=models.SET_NULL, null=True, blank=True
```
Comment (odinuge) [inline lego/apps/pinned/models.py]: Maybe cascade these, so we don't end up with `pinned` values without an event or article?

File: lego/apps/pinned/views.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+class PinnedViewSet(AllowedPermissionsMixin, viewsets.ModelViewSet):
+
+    permission_classes = (IsAuthenticated, )
+    serializer_class = CreatePinnedSerializer
+    queryset = Pinned.objects.all()
+
+    @list_route(methods=['GET'])
+    def mine(self, request, pk=None):
+        queryset_events = Pinned.objects.filter(
```
Comment (odinuge) [inline lego/apps/pinned/views.py]: Have you tested how many queries these create?

File: lego/apps/pinned/views.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+from lego.apps.permissions.api.views import AllowedPermissionsMixin
+from lego.apps.pinned.models import Pinned
+from lego.apps.pinned.serializers import (
+    CreatePinnedSerializer, PinnedArticleSerializer, PinnedEventSerializer
+)
+
+
+class PinnedViewSet(AllowedPermissionsMixin, viewsets.ModelViewSet):
+
+    permission_classes = (IsAuthenticated, )
```
Comment (odinuge) [inline lego/apps/pinned/views.py]: :trollface:

**Comment by odinuge** [comment]
What is the status on this?? :man_shrugging: Quite far behind master already... :/

**Comment by JonasBak** [comment]
> What is the status on this??  Quite far behind master already... :/

I got a bit stuck working on the frontend as i could not find any ways of showing what is pinned that actually improves the current system from a users perspective. The biggest issue with the current system is that when more than one item is pinned, only one of them shows up as pinned on the frontpage, and i found it hard to improve on this without adding too much clutter to the frontpage. I've talked a bit with others about how to solve this, but will probably need more help.

As for the conflicts; none of the conflicting files are actually that important, so I don't think it will be that much work rebasing.

**Pull request closed without merging**